resources/ts/Pages/Integrations/New.tsx

```diff
@@ -77,6 +67,17 @@ const New = ({ subscriptions }: Props) => {
   const { data, setData, errors, hasErrors, post, processing } =
     useForm(initialFormValues);
 
+  useEffect(() => {
+    const freeSubscriptionId = subscriptions.find(
+      (subscription) =>
+        subscription.integrationType === activeType &&
+        subscription.category === SubscriptionCategory.Free
+    )?.id;
+
+    setData("subscriptionId", freeSubscriptionId ?? "");
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [activeType, subscriptions]);
```

Comment:
Is setData the missing dependency?

Reply:
Yeah it's the same issue as for all the form related methods they seem to completely mess with useEffect, but I wasn't sure if it was worth adding the same comment about it as in the other files :/ Im not sure if we're doing something wrong with useForm or something